apps/freeablo/faworld/playerbehaviour.cpp

```diff
@@ -104,7 +107,10 @@ namespace FAWorld
             case PlayerInput::Type::CastSpell:
             {
                 auto clickedPoint = Misc::Point(input.mData.dataCastSpell.x, input.mData.dataCastSpell.y);
-                mPlayer->castActiveSpell(clickedPoint);
+                if (!cursorItem.isEmpty())
```

Comment:
This doesn't make sense to me, there's a separate PlayerInput for dealing with dropping items.

Reply:
Ahh I see, `LocalInputHandler` should do this check. Yep that would make this much clearer.

Also since adding this "cast spell input" for right click, I've found out that it's incorrect. Right click doesn't mean cast spell, it means "do whatever action you've configured it to do" (via the bottom ctrl panel). So there's a bit of refactoring needed here as well